### PR TITLE
[nginx-ingress-controller] Fix duplicated real_ip_header

### LINKED ISSUE #1316: [nginx-ingress-controller] real_ip_header is duplicate when useProxyProtocol is enabled
It seems as though since https://github.com/kubernetes/contrib/commit/a152cc44707b8f214666ba7468ceb6575478333b, if useProxyProtocol is enabled, real_ip_header is set twice causing an error when reloading nginx: 

```
nginx: [emerg] "real_ip_header" directive is duplicate in /etc/nginx/nginx.conf:65
```

This prevents the ingress controller from starting in 0.8 if useProxyProtocol is enabled.

### PR DESCRIPTION
**Pull request by aledbf**
fixes #1316

### CODE REVIEW

File: ingress/controllers/nginx/nginx.tmpl [hunk 1]
```diff
@@ -17,8 +17,14 @@ events {
 
 http {
     {{/* we use the value of the header X-Forwarded-For to be able to use the geo_ip module */}}
+    {{ if $cfg.useProxyProtocol -}}
+    set_real_ip_from    {{ $cfg.proxyRealIpCidr }};
+    real_ip_header      proxy_protocol;
```
Comment (bprashanth) [inline ingress/controllers/nginx/nginx.tmpl]: for my own eductaion, this actually sets the ip on the forwarded packet from the proxy protocol ascii block, right?

**Comment by bprashanth** [comment]
I thought we were doing some static checking of the config before reload (and not reloading if the config was invalid)? if that's not the case, can you please file a follow up bug. LGTM.

File: ingress/controllers/nginx/nginx.tmpl (hunk 1 above)
Comment (aledbf) [inline ingress/controllers/nginx/nginx.tmpl]: yes

**Comment by aledbf** [comment]
If reload returns an error nginx continues using the previous version

**Comment by bprashanth** [comment]
ah so this won't cause downtime right? just a broken feature

**Comment by aledbf** [comment]
exactly. This is an issue when you start with an error in nginx.conf not in a reload

**Comment by bprashanth** [comment]
hmm, we can probably work around that too by persisting the last known good template in a volume of configmap, reading on boot up, and only overwriting it on success reload. But a better first step would be an e2e for the nginx controller :)

**Pull request merged**